contracts/RMRK/utils/RMRKNestableUtils.sol

```diff
@@ -0,0 +1,122 @@
+     */
+    function validateChildOf(
+        address parentAddress,
+        address childAddress,
+        uint256 parentId,
+        uint256 childId
+    ) public view returns (bool) {
+        if (
+            !IERC165(childAddress).supportsInterface(type(IERC6059).interfaceId)
+        ) {
+            return false;
+        }
+
+        (address directOwner, uint256 ownerId, ) = IERC6059(childAddress)
+            .directOwnerOf(childId);
+
+        if (directOwner == parentAddress && ownerId == parentId) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @notice Used to validate whether the specified child token is owned by a given parent token.
+     * @param parentAddress Address of the parent token's collection smart contract
+     * @param childAddresses An array of the child token's collection smart contract addresses
+     * @param parentId ID of the parent token
+     * @param childIds An array of child token IDs to verify
+     * @return A boolean value indicating whether all of the child tokens are owned by the parent token or not
+     * @return An array of smart contract addresses of the tokens that are not owned by the parent token
+     * @return An array of token IDs of child tokens that are not owned by the parent token
+     */
+    function validateChildrenOf(
+        address parentAddress,
+        address[] memory childAddresses,
+        uint256 parentId,
+        uint256[] memory childIds
+    ) public view returns (bool, address[] memory, uint256[] memory) {
+        if (childAddresses.length != childIds.length) {
+            revert RMRKMismachedArrayLength();
+        }
+
+        address directOwner;
+        uint256 ownerId;
+        uint256 numberOfInvalidChildTokens;
+        address[] memory tmpInvalidChildAddresses = new address[](
+            childAddresses.length
+        );
+        uint256[] memory tmpInvalidChildIds = new uint256[](
+            childAddresses.length
+        );
+        bool isValid = true;
+
+        for (uint256 i; i < childAddresses.length; ) {
+            if (
+                IERC165(childAddresses[i]).supportsInterface(
+                    type(IERC6059).interfaceId
+                )
+            ) {
+                (directOwner, ownerId, ) = IERC6059(childAddresses[i])
+                    .directOwnerOf(childIds[i]);
+            }
+
+            if (directOwner != parentAddress || ownerId != parentId) {
+                tmpInvalidChildAddresses[
+                    numberOfInvalidChildTokens
+                ] = childAddresses[i];
+                tmpInvalidChildIds[numberOfInvalidChildTokens] = childIds[i];
+                numberOfInvalidChildTokens++;
+                isValid = false;
+            }
+
+            delete directOwner;
+            delete ownerId;
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        address[] memory invalidChildAddresses = new address[](
```

Comment:
What if you simple returned an array of bools indicating for each child, in the same order, whether it was valid or not? 
Would be much cheaper and should be equally useful to the front.